### PR TITLE
Generate Gaussian Smooth data labels via App-wide data labeler

### PR DESCRIPTION
**Pull request by duytnguyendtn**
### Description

This is a quick PR to fix a (potential?) bug where the first smoothed dataset generated by the Gaussian Smooth plugin is missing the dataset name. As requested by Cami, all smoothed datasets should have consistent naming by having the original dataset represented in its name. I replaced the custom labeling logic with Jesse's app-wide data label generator for consistency.

Before:
image

After:
image

This is due to some weird (potentially buggy) behavior where the dataset field is unavailable until multiple "smoothable" data products are available:

image

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py [hunk 1]
```diff
@@ -88,15 +88,19 @@ def user_api(self):
...
-        if self.config == "cubeviz":
-            label_comps += [f"{self.mode_selected.lower()}-smooth"]
+        '''Generate a label and set the results field to that value'''
+        if (hasattr(self, 'dataset') and (len(self.dataset.labels) >= 1) or self.app.config == 'mosviz'):  # noqa
+            dataset = self.dataset_selected
         else:
-            label_comps += ["smooth"]
-        label_comps += [f"stddev-{self.stddev}"]
-        self.results_label_default = " ".join(label_comps)
+            dataset = ''
```
Comment (kecnry) [inline jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py:95]: won't this result in an empty space at the beginning of the formatted label?

File: jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py [hunk 2]
```diff
@@ -88,15 +88,19 @@ def user_api(self):
...
+
+        smooth_type = (f"{self.mode_selected.lower()}-smooth" if self.config == "cubeviz"
+                       else "smooth")
+        stddev = f"stddev-{self.stddev}"
+
+        # Overriding is allowed, so do not check for uniqueness
+        self.results_label_default = (
+            self.app.return_data_label(f"{dataset} {smooth_type} {stddev}", check_unique=False))
```
Comment (kecnry) [inline jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py:103]: what benefit does passing this through `return_data_label` provide?

**Comment by kecnry** [comment]
I also think this probably should have a changelog entry since it is a change in behavior.  (And it wasn't really a bug - this was originally by design and I think we switched individual plugins later by request... are there any other that need updates to be consistent?)

File: jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py (hunk 1 above)
Comment (duytnguyendtn) [reply jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py:95]: I traced the behavior, and the path is strange... let me try to explain (in Cubeviz):

When a user loads their data (which happens usually at the start of their environment), this code is triggered 3 times: (1) When the config is initialized, (2) when data is added to the flux viewer, and (3) adding the autocollapsed spectrum to the spectrum viewer. The first two times, `self.dataset` is empty, which, yes, sets the empty string and results in an empty space at the start. But the third time, the dataset is registered with the plugin and it properly populates `self.dataset` which then fills the text field automatically. 

All of this is transparent to the user, since it's all happening at initialization time. The only time the user might find this scenario is if they start the app and open the plugin, but haven't loaded any data into the tool, in which case they can't even smooth to begin with.

To put it another way, testing against both Cubeviz and Specviz, the only time I could see the empty space is when there is no data in the tool at all. In which case trying to smooth results in a traceback (unsurprisingly). Once I loaded data, the string formatted properly.

File: jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py (hunk 2 above)
Comment (duytnguyendtn) [reply jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py:103]: In this Gaussian Smooth case, not much. This is because Gaussian Smooth allows the user to overwrite the result. By putting `check_unique=False`, it effectively just returns the string.

The main reason I decided to pass it through anyways is mainly for consistency. Though this is the case with Gaussian Smooth, it may not be the case with other plugins. We should probably centralize our data label generation, especially if we want automatic uniqueness checks. So I pass it through because (1) I don't think it will cause any harm and (2) it can serve as a reminder for future development to follow this paradigm.

I can also just set it if we don't want to

**Review by pllim — COMMENTED** [review]
I agree with Kyle that change log is necessary here.

I am pretty neutral on the new naming convention. I think POs will be the picky ones here.

**Comment by pllim** [comment]
Current CI failure is unrelated, though I do wonder if any tests would fail due to this change here.

File: jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py (hunk 2 above)
Comment (kecnry) [reply jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py:103]: @javerbukh was the `return_data_label` intended to be used for plugin labels?  If so, I'd agree it makes sense to start introducing this, but otherwise I'm not sure its worth the overhead since plugin labels have their own logic to check for overwriting, etc.

**Review by camipacifici — APPROVED** [review]
Does what I asked :) thank you!

**Comment by pllim** [comment]
I restarted the CI 🤞

File: jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py (hunk 1 above)
Comment (kecnry) [reply jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py:95]: ok, that makes sense.  I realize it's out of scope of what was asked here, but how do you feel about just returning an empty string in that case (for the whole label)?  That might be easier to follow in the future.  Or alternatively an inline comment in the else explaining that it will be immediately rewritten once data is loaded?

Comment (duytnguyendtn) [reply jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py:95]: All the above being said, I think having _something_ in the field is also a good "but what if" backup scenario. In the event something goes off the rails and a dataset is _somehow_ created with these strings, at least it's descriptive enough to know _kind of_ what it is. I'm nervous with it being entirely blank. Thoughts?

File: jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py (hunk 2 above)
Comment (duytnguyendtn) [reply jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py:103]: For more context, I originally thought I could provide the smoothing details (e.g. `spatial-smoothed stddev 1.0`) as the `ext` argument, but because we allow overwriting (i.e. don't enforce uniqueness), we can't; `ext` is only used if uniqueness is required

File: jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py (hunk 1 above)
Comment (kecnry) [reply jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py:95]: ok, that's a good point, I think this is probably fine for now and if we ever run into a case where its a problem we can update it then.  Thanks!

Comment (duytnguyendtn) [reply jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py:97]: Randomly came up with a potential solution to your point here in https://github.com/spacetelescope/jdaviz/pull/1973/commits/33f577a5ab85d1244d5eccaf13f350b758273bf8. This should avoid the leading whitespace

File: jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py (hunk 2 above)
Comment (javerbukh) [reply jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py]: The first pass implementation did not include plugin labels because I believe there is already a mixin in utils that handles that (AddResult maybe?). I agree with @duytnguyendtn that I do not see the harm in doing it this way, especially since we may want to centralize all data label generation in the future, but I also see @kecnry point about unnecessary overhead (though if we're not checking for uniqueness it should always be a fast operation). So I don't think I can weigh in too heavily on one side or the other here.

**Review by kecnry — APPROVED** [review]
I would slightly vote to either remove the `app.return_data_label` call or use it consistently across all plugins, but that doesn't need to block approval/merge if anyone disagrees and would prefer we just slowly transition to using that as we come across them.
